docker/stream-minio/stream-minio.js

```diff
@@ -72,14 +78,25 @@ const streamFileToMinio = (url, objectMap = {}) => {
     if (!hasBucket) await minioClient.makeBucket(BUCKET_NAME, REGION);
 
     const objectMap = await getObjectMap();
-    await asyncForEach(FILE_URLS, url => streamFileToMinio(url, objectMap));
+    console.log(`received ${FILE_URLS.length} urls to upload to MinIO.`)
+    const skippedUrls = FILE_URLS.filter((url) => path.basename(url) in objectMap);
+    for (const url of skippedUrls) {
+      console.log(`skipping ${url}`);
+    }
+    const urlsToUpload = FILE_URLS.filter((url) => !(path.basename(url) in objectMap));
+    // We use asyncForEach instead of Promise.all to avoid overloading the MinIO server
+    // Ideally we'd have a queue system that limits the number of simultaneous uploads
+    const uploadedObjects = await asyncForEach(urlsToUpload, (url) => streamFileToMinio(url));
+    await writeXCom({uploadedObjects, skippedUrls});
   } catch (err) {
     console.error(err);
   }
 })();
 
 async function asyncForEach(array, callback) {
```

Comment:
it looks great!
can we just change the name from `asyncForEach` to `asyncMap` something like this to give more intuitive idea to users? since js `[].forEach` doesn't have return values but `[].map` has

Reply:
sounds good!